gobblin-runtime/src/main/java/gobblin/runtime/locks/JobLock.java

```diff
@@ -27,40 +28,51 @@
  *
  * @author Yinan Li
  */
-public interface JobLock {
+public interface JobLock extends Closeable {
+
+  /**
+   * Initializes the lock.
+   *
+   * @param properties  the job properties
+   * @param jobLockEventListener the listener for lock events
+   * @throws JobLockException thrown if the {@link JobLock} fails to initialize
+   */
+  void initialize(Properties properties, JobLockEventListener jobLockEventListener)
```

Comment:
I don't think this can be moved to the constructor as we have no way to ensure that implementations still provide a constructor with those arguments.  As a compromise, I will add an `abstract` class called `ConfigurableJobLock` with the `void configure(Properties properties, JobLockEventListener jobLockEventListener)` method.  The factory will check if the `JobLock` extends `ConfigurableJobLock ` and if so it will call the `configure` method.